app/frontend/src/features/profile/actions/PendingRequest.tsx

```diff
@@ -0,0 +1,61 @@
+/* has to check whether request is sent or received
+if sent, return a simple disabled "pending request" button
+if recieved, return the response button*/
+import { SetMutationError } from "features/connections/friends";
+import useFriendRequests from "features/connections/friends/useFriendRequests";
+import React from "react";
+
+import PendingFriendRequestSent from "./PendingFriendRequestSent";
+import RespondToFriendRequestProfileButton from "./RespondToFriendRequestProfileButton";
+
+interface PendingRequestProps {
+  userId: number;
+  setMutationError: SetMutationError;
+}
+
+function CheckSentRequests(userId: number) {
+  const friendRequestsSent = useFriendRequests("sent").data;
+  const friendRequestToUser = friendRequestsSent?.find(
+    (e) => e.userId === userId
+  );
+  if (friendRequestToUser !== undefined) {
+    const request = friendRequestToUser;
+    return request;
+  } else {
+    return false;
+  }
+}
+
+function CheckReceivedRequests(userId: number) {
+  const { data } = useFriendRequests("received");
+  const friendRequestsReceived = data;
+  const friendRequestFromUser = friendRequestsReceived?.find(
+    (e) => e.userId === userId
+  );
+  if (friendRequestFromUser !== undefined) {
+    const request = friendRequestFromUser;
+    return request;
+  } else {
+    return false;
+  }
+}
```

Comment:
These functions should be hooks. `useRequestSentToUser` / `useRequestReceivedFromUser` (or a better name, if you can think of one). The reason is they contain hooks themselves, and hooks and only be in other hooks or components (you probably had a lint error which went away when you put a capital letter. Sentence case is for components only, and hooks are allowed in components)